core/server/master/src/main/java/alluxio/master/file/DefaultFileSystemMaster.java

```diff
@@ -4843,6 +4811,64 @@ public void heartbeat() throws InterruptedException {
     public void close() {} // Nothing to clean up.
   }
 
+  private final class AsyncInodeStoreLoadTask implements Runnable {
+    @Override
+    public void run() {
+      int numToBePersisted = mInodeTree.getToBePersistedIds().size();
+      int actualToBePersisted = 0;
+      long start = System.currentTimeMillis();
+      LOG.info("[AE-2584] start fs master start time, numToBePersisted:{}", numToBePersisted);
+
+      // Rebuild the list of persist jobs (mPersistJobs) and map of pending persist requests
+      // (mPersistRequests)
+      long persistInitialIntervalMs =
+              Configuration.getMs(PropertyKey.MASTER_PERSISTENCE_INITIAL_INTERVAL_MS);
+      long persistMaxIntervalMs =
+              Configuration.getMs(PropertyKey.MASTER_PERSISTENCE_MAX_INTERVAL_MS);
+      long persistMaxWaitMs =
+              Configuration.getMs(PropertyKey.MASTER_PERSISTENCE_MAX_TOTAL_WAIT_TIME_MS);
+
+      for (Long id : mInodeTree.getToBePersistedIds()) {
```

Comment:
Normally the inodes are accessed under locks. I think there could be some safety issues here if the jobs are added without taking the lock on the individual inode). You may be able to lock the inode individually here for safety (normally they are locked by locking a path, but it should be possible to just lock them individually)?

Reply:
For these 2 types of inodes:
  1) with JobId -> add to mPersistRequests, and scheduler will generate new jobid from submitting to job master
  2) w/o. JobId -> this is from previous submitted job id, and only need to add to mPersistJobs

Others who interact with these 2 concurrent map apart from the periodic task PersistenceScheduler/Checker:
   - for mPersistRequests : delete (WRITE_INODE full path lock) / rename (WRITE_EDGE) / scheduleAsyncPersistenceInternal (from scheduleAsyncPersistenceInternal and completeFile both with WRITE_INODE) 
  - for mPersistJobs : delete (WRITE_INODE) 
These are guarded under a full path lock, so put a full path lock of READ to leverage the same